File: ContainerWithMostWater/containerWithMostWater.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
  let left;
  let highestSolution = 0;
  let current_highest = 0;;
  for (let i = 0; i < height.length - 1; i++) {
    left = i + 1;
    while (left < height.length) {
      current_highest = (Math.min(height[i], height[left])) * (left - i);
      highestSolution = current_highest > highestSolution ? current_highest : highestSolution;
      left++;
    }
  }
  return highestSolution;
};


const x = [
  [1,8,6,2,5,4,8,3,7],
  [1,1]
];

x.forEach(element => {
  console.log(maxArea(element));
});

File: ContainerWithMostWater/containerWithMostWater.py

class Solution:
    def maxArea(self, height) -> int:
        left = 0
        right = len(height) - 1
        maxArea = 0
        while right > left:
            currentArea = (min(height[left], height[right])) * (right - left)
            if currentArea > maxArea:
                maxArea = currentArea
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        return maxArea
    

height = [1,8,6,2,5,4,8,3,7]
sol = Solution()
print(sol.maxArea(height))